Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -3529,21 +3529,29 @@ private async Task<PipExecutionStep> ExecutePipStep(RunnablePip runnablePip)
             }
         }
 
-        private static void FlagSharedOpaqueOutputs(IPipExecutionEnvironment environment, ExecutionResult executionResult)
+        private void FlagSharedOpaqueOutputs(IPipExecutionEnvironment environment, ProcessRunnablePip process)
         {
+            // Select all declared output files
+            IEnumerable<(AbsolutePath path, bool isKnownToBeUnderSharedOpaque)> paths = process.Process
+                .FileOutputs
+                .Select(f => (path: f.Path, isKnownToBeUnderSharedOpaque: false));

```

Comment:
>false [](start = 74, length = 5)

Is this actually always false? Thinking in the case where a pip may declare a file under one of its own shared opaques